client/ayon_core/pipeline/workfile/path_resolving.py

```diff
@@ -278,42 +273,141 @@ def get_last_workfile_with_version(
 
     # Get highest version among existing matching files
     version = None
-    output_filenames = []
-    for filename in sorted(filenames):
+    output_filepaths = []
+    for filepath in sorted(filepaths):
+        filename = os.path.basename(filepath)
         match = re.match(file_template, filename, **kwargs)
         if not match:
             continue
 
         if not match.groups():
-            output_filenames.append(filename)
+            output_filepaths.append(filename)
             continue
 
         file_version = int(match.group(1))
         if version is None or file_version > version:
-            output_filenames[:] = []
+            output_filepaths.clear()
             version = file_version
 
         if file_version == version:
-            output_filenames.append(filename)
+            output_filepaths.append(filepath)
+
+    output_filepath = None
```

Comment:
This really warrants a comment I noticed.
```suggestion
    # Among matching (highest) version numbers 
    # take the last modified file
    output_filepath = None
```